### PR TITLE
Add interfaces

### PR DESCRIPTION
**Pull request by kanitw**
## TODO: later
- [ ] things in schema are actually constants — we should eventually use ALL CAPS?
- [ ] declare interface for Config (once config is more stable)

### CODE REVIEW

File: src/schema/axis.schema.ts [hunk 1]
```diff
@@ -1,3 +1,18 @@
+export interface Axis {
```
Comment (domoritz) [inline src/schema/axis.schema.ts]: FYI, we should call this IAxis. Seems to be the standard.

Comment (domoritz) [inline src/schema/axis.schema.ts]: `interface-name enforces the rule that interface names must begin with a capital 'I'`

https://github.com/palantir/tslint

Comment (domoritz) [inline src/schema/axis.schema.ts]: Discussed offline.

**Comment by domoritz** [comment]
fielddef.schema missing @kanitw

**Comment by kanitw** [comment]
@domoritz Can you clarify that?

**Comment by domoritz** [comment]
The tests are failing and this PR is not marked WIP.

**Comment by kanitw** [comment]
If you test that offline, do you have the same problem?   I don't know how to reproduce that.  Test are passing in my machine.

**Comment by domoritz** [comment]
Hmm, odd. No, don't see that issue.

File: src/schema/schema.ts [hunk 2]
```diff
@@ -1,305 +1,35 @@
...
+export interface Spec {
+  data?: Data;
+  marktype?: string;
+  encoding?: Encoding;
+  config?: any; // FIXME: declare
+}
 
-export {aggregate} from './fielddef.schema';
+// TODO remove this
+export {aggregate} from './fieldDef.schema';
```
Comment (domoritz) [inline src/schema/schema.ts]: OSX is weird and does not respect capitalization. You need to fix this!

**Comment by kanitw** [comment]
Done

**Pull request merged**